holepunch: flaky TestFailuresOnInitiator

https://github.com/libp2p/go-libp2p/actions/runs/3654057481/jobs/6174141473

```
  === RUN   TestFailuresOnInitiator/no_addrs_after_filtering
      holepunch_test.go:241: 
          	Error Trace:	/Users/runner/work/go-libp2p/go-libp2p/p2p/protocol/holepunch/holepunch_test.go:241
          	Error:      	"failed to open hole-punching stream: protocol not supported" does not contain "aborting hole punch initiation as we have no public address"
          	Test:       	TestFailuresOnInitiator/no_addrs_after_filtering
  --- FAIL: TestFailuresOnInitiator (0.51s)
      --- PASS: TestFailuresOnInitiator/responder_does_NOT_support_protocol (0.09s)
      --- PASS: TestFailuresOnInitiator/responder_does_NOT_send_a_CONNECT_message (0.12s)
      --- PASS: TestFailuresOnInitiator/unable_to_READ_CONNECT_message_from_responder (0.10s)
      --- PASS: TestFailuresOnInitiator/responder_does_NOT_reply_within_hole_punch_deadline (0.11s)
      --- FAIL: TestFailuresOnInitiator/no_addrs_after_filtering (0.09s)

--- a/p2p/protocol/holepunch/holepunch_test.go
+++ b/p2p/protocol/holepunch/holepunch_test.go
@@ -186,13 +186,12 @@ func TestFailuresOnInitiator(t *testing.T) {
 		},
 		"responder does NOT reply within hole punch deadline": {
 			holePunchTimeout: 10 * time.Millisecond,
-			rhandler: func(s network.Stream) {
-				time.Sleep(5 * time.Second)
-			},
-			errMsg: "i/o deadline reached",
+			rhandler:         func(s network.Stream) { time.Sleep(5 * time.Second) },
+			errMsg:           "i/o deadline reached",
 		},
 		"no addrs after filtering": {
-			errMsg: "aborting hole punch initiation as we have no public address",
+			errMsg:   "aborting hole punch initiation as we have no public address",
+			rhandler: func(s network.Stream) { time.Sleep(5 * time.Second) },
 			filter: func(remoteID peer.ID, maddrs []ma.Multiaddr) []ma.Multiaddr {
 				return []ma.Multiaddr{}
 			},

